### PR TITLE
Show confirm modal with merchant info on configuration before saving new credentials.

### PR DESCRIPTION
**Pull request by pniel-cohen**
Show confirm modal with merchant info on configuration before saving new credentials.

LR-7534 Improve visibility to prevent wrong connections between Yotpo L&R admin to Magento Platform

image

### CODE REVIEW

File: Block/Adminhtml/System/Config/MerchantInfoConfirm.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<?php
```
Comment (shalomgu) [inline Block/Adminhtml/System/Config/MerchantInfoConfirm.php:1]: With new PR we bump the version number, aren't we?

**Review by shalomgu — COMMENTED** [review]
Looks good
I would be happy to do 30 min review together to refresh my MagentoPhp mem

File: Block/Adminhtml/System/Config/MerchantInfoConfirm.php (hunk 1 above)
Comment (pniel-cohen) [reply Block/Adminhtml/System/Config/MerchantInfoConfirm.php:1]: It depends if you want to include more items on the release or consider this change a new version on it's own.
Technically it's a choice and personally I prefer to separate the version bumping from the actual change (most of the times).
We'll discuss that internally :)

File: Block/Adminhtml/System/Config/MerchantInfoConfirm.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+     * Remove scope label
+     *
+     * @param  AbstractElement $element
+     * @return string
+     */
+    public function render(AbstractElement $element)
+    {
+        $element->unsScope()->unsCanUseWebsiteValue()->unsCanUseDefaultValue();
+        return parent::render($element);
+    }
```
Comment (shalomgu) [inline Block/Adminhtml/System/Config/MerchantInfoConfirm.php:65]: I'll need explanation on this (can do zoom to save time)

File: Block/Adminhtml/System/Config/MerchantInfoConfirm.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    /**
+     * Return element html
+     *
+     * @param  AbstractElement $element
+     * @return string
+     */
+    protected function _getElementHtml(AbstractElement $element)
+    {
+        return $this->_toHtml();
```
Comment (shalomgu) [inline Block/Adminhtml/System/Config/MerchantInfoConfirm.php:75]: Also to this :-)

File: Block/Adminhtml/System/Config/MerchantInfoConfirm.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    public function getMerchantInfoApiUrl()
+    {
+        return AbstractApi::SWELL_API_PRODUCTION_URL . 'merchant_info';
```
Comment (shalomgu) [inline Block/Adminhtml/System/Config/MerchantInfoConfirm.php:80]: . is to concat ?

File: Block/Adminhtml/System/Config/MerchantInfoConfirm.php [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    public function getSwellGuid()
+    {
+        if ($this->_storeId !== null) {
+            return $this->_yotpoHelper->getSwellGuid(ScopeInterface::SCOPE_STORE, $this->_storeId);
+        } elseif ($this->_websiteId !== null) {
+            return $this->_yotpoHelper->getSwellGuid(ScopeInterface::SCOPE_WEBSITE, $this->_websiteId);
+        } else {
+            return $this->_yotpoHelper->getSwellGuid();
+        }
+    }
```
Comment (shalomgu) [inline Block/Adminhtml/System/Config/MerchantInfoConfirm.php:92]: Dont we have this in a shared place ?

File: Block/Adminhtml/System/Config/MerchantInfoConfirm.php [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    public function getRootApiUrl()
+    {
+        if ($this->_storeId !== null) {
+            return rtrim($this->_yotpoHelper->getBaseUrl(ScopeInterface::SCOPE_STORE, $this->_storeId), "/") . "/rest/V1";
+        } elseif ($this->_websiteId !== null) {
+            return rtrim($this->_yotpoHelper->getBaseUrl(ScopeInterface::SCOPE_WEBSITE, $this->_websiteId), "/") . "/rest/V1";
+        } else {
+            return rtrim($this->_yotpoHelper->getBaseUrl(), "/") . "/rest/V1";
+        }
+    }
```
Comment (shalomgu) [inline Block/Adminhtml/System/Config/MerchantInfoConfirm.php:103]: Dont we have this in a shared place ?

**Review by shalomgu — APPROVED** [review]

**Review by shalomgu — APPROVED** [review]
Looks great.
Per the refactor I guess you'll do a bit more testing on the code that changed 👍 
Cheers

**Pull request merged**